python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1875,6 +1879,96 @@ def get_event_records(
         """
         return self._event_storage.get_event_records(event_records_filter, limit, ascending)
 
+    @public
+    @traced
+    def get_materialization_records(
+        self,
+        records_filter: Optional[Union[AssetKey, "AssetRecordsFilter"]] = None,
+        limit: Optional[int] = None,
+        ascending: bool = False,
+    ) -> Sequence["EventLogRecord"]:
+        """Return a list of materialization records stored in the event log storage.
+
+        Args:
+            records_filter (Optional[Union[AssetKey, AssetRecordsFilter]]): the filter by which to
+                filter event records.
+            limit (Optional[int]): Number of results to get. Defaults to infinite.
```

Comment:
Unless you have a good case against it, this type of api change seems like a good chance to correct the unscalable choice of unbounded limits on these calls. Some sort of env var overridable max events that we default to / ensure the explicit limit is under. 10k is current implicit get_event_records limit in cloud

Reply:
The main worry I have here is changing all the call sites within the dagster framework to use it with limits...  I think there are a handful of places where we don't, but I guess this is a good opportunity to switch.